libs/shared/ui-components/src/Form/Form.tsx

```diff
@@ -32,8 +32,20 @@ export const Form: FC<TFormProps> = (props) => {
   const {
     register,
     handleSubmit,
+    setValue,
     formState: { errors, isValid },
-  } = useForm<FormValues>({ mode: 'onChange' });
+  } = useForm<FormValues>({ mode: 'onSubmit' });
```

Comment:
If I read the docs right, this change should make it so that it only validates form contents on submit attempts?

If so, something is weird, though not necessarily a merge blocker. After I attempt to submit the form with 1+ fields empty, I get the red "This field is required" message under the empty ones. Then, if I add content to one of them, the red "This field is required" goes away immediately, without my clicking 'Submit'. If I re-delete that new content, the red "This field is required" returns immediately, without my clicking 'Submit'.

Is it perhaps the case that `mode: 'onChange'` behavior is enabled after a failed form submit?

Reply:
Yeah, I think the part of the react-hook-form API that explains what you're seeing is an option called `reValidateMode` (default is `onChange`).

Also, this change slipped through the cracks. My first attempt at fixing the bug was to change the mode from `onChange` to `onSubmit`, but that didn't do the trick, and I realized that there was probably no easy way to get the Form component to re-validate the form after I changed a value via `messageField.value = msg`. 

I can revert this change if you prefer.